homeassistant/components/tuya/__init__.py

```diff
@@ -134,6 +136,55 @@ async def async_remove_entry(hass: HomeAssistant, entry: TuyaConfigEntry) -> Non
     await hass.async_add_executor_job(manager.unload)
 
 
+class TuyaManager(Manager):
+    """Patched tuya_sharing Manager.
+
+    This can be removed once change has been merged upstream in tuya-device-sharing-sdk. Please
+    check the status of https://github.com/tuya/tuya-device-sharing-sdk/pull/17
+    """
```

Comment:
This is not code that should live in our codebase. We have requirements on things like these that they have to live in an separate package.

../Frenck

Reply:
Hmm... the last commit on https://github.com/tuya/tuya-device-sharing-sdk is from 9 months ago. None of the recent pull requests on that repo have received any response (https://github.com/tuya/tuya-device-sharing-sdk/pulls).

Which option do you suggest to move forward:

1. fork the entire code from https://github.com/tuya/tuya-device-sharing-sdk/tree/main/tuya_sharing into this component, i.e. move it under `homeassistant/components/tuya/...`
2. fork `tuya-device-sharing-sdk` and replace the dependency from this component with that new package
3. create a new library which just contains just this patch and add it as an additional dependency to this component (so it would have 2 dependencies going forward)

IMO 1) would be the best way forward as it allows the community here to maintain this code instead of continuing to rely on a unmaintained library.

If we go with 3), where should that code live and who should own that library?

Reply:
Did you contact Tuya?

Reply:
PS: The upstream PR you linked in the description is from yesterday... 🤷 

Like that is not much time you give to something open source.

Reply:
Yeah, I sent mine just yesterday but there are two others with one being a trivial 2 line change from November 2023 that hasn't been looked at. 

I haven't contacted Tuya as I don't have any contact there. Do you?